File: shanny-blog-java/blog-common/src/main/java/com/shanny/result/Result.java
package com.shanny.result;

import com.shanny.enums.ResultEnum.*;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.io.Serializable;

@Data
@Schema(description = "状态返回")
public class Result<T> implements Serializable {
    @Schema(description = "状态码")
    private ResultCode code;

    @Schema(description = "错误信息")
    private String msg;

    @Schema(description = "返回数据")
    private T data;

    public static <T> Result<T> success() {
        Result<T> result = new Result<>();
        result.code = ResultCode.SUCCESS;
        return result;
    }
    public static <T> Result<T> success(String msg) {
        Result<T> result = new Result<>();
        result.msg = msg;
        result.code = ResultCode.SUCCESS;
        return result;
    }

    public static <T> Result<T> success(T object) {
        Result<T> result = new Result<>();
        result.data = object;
        result.code = ResultCode.SUCCESS;
        return result;
    }

    public static <T> Result<T> success(String msg, T object) {
        Result<T> result = new Result<>();
        result.msg = msg;
        result.data = object;
        result.code = ResultCode.SUCCESS;
        return result;
    }

    public static <T> Result<T> error(String msg) {
        Result<T> result = new Result<T>();
        result.msg = msg;
        result.code = ResultCode.ERROR;
        return result;
    }
}

File: shanny-blog-net/blog-common/Result/Result.cs
using blog_common.Constant;

namespace blog_common.Result
{
    public class Result<T>
    {
        public string Code { get; set; }
        public string Msg { get; set; } = string.Empty;
        public T? Data { get; set; }

        public static Result<T> Success()
        {
            return new Result<T>
            {
                Code = ResultMsg.CODE_SUCCESS
            };
        }

        public static Result<T> Success(string msg)
        {
            return new Result<T>
            {
                Code = ResultMsg.CODE_SUCCESS,
                Msg = msg
            };
        }

        public static Result<T> Success(T? data)
        {
            return new Result<T>
            {
                Code = ResultMsg.CODE_SUCCESS,
                Data = data
            };
        }

        public static Result<T> Success(string msg, T? data)
        {
            return new Result<T>
            {
                Code = ResultMsg.CODE_SUCCESS,
                Msg = msg,
                Data = data
            };
        }

        public static Result<T> Error(string msg)
        {
            return new Result<T>
            {
                Code = ResultMsg.CODE_FAIL,
                Msg = msg
            };
        }
    }
}